pylint/checkers/format.py

```diff
@@ -602,9 +602,34 @@ def _closing_bracket(self, tokens, i):
 
         self._check_space(tokens, i, (policy_before, _IGNORE))
 
+    def _has_valid_type_annotation(self, tokens, i):
+        """Extended check of PEP-484 type hint presence"""
+        if not self._inside_brackets('('):
+            return False
+        pos = i - 1
+        bracket_level = 0
+        while pos:
```

Comment:
I think this would be better as a for loop over a reverse slice:  `for token in tokens[::-1]:`